### DISCUSSION TITLE
Shift+Enter or Ctrl+Enter to add a new paragraph

### DISCUSSION THREAD

**Discussion by pablowatson** [ueberdosis/tiptap#804] [General Questions]
Hi,

I wanted to disable enter such that it does not create a new paragraph (which I was able to figure out how to do), but I would also like to add new paragraphs  using enter + some combination of modifier key (either ctrl or shift). I thought I could modifiy the HardBreak node to achieve this, but I cannot quite get the results that I want. e.g.

```
class ShifEnterParagraph extends Node {

  public get name() {
    return 'shiftenterparagraph';
  }

  public get schema() {
    return {
      inline: true,
      group: 'inline',
      selectable: false,
      parseDOM: [{ tag: 'p' }],
      toDOM: () => ['p'],
    };
  }

  public keys({ type }: { type: any }) {
    const command = chainCommands(exitCode, (state: any, dispatch: any) => {
      dispatch(state.tr.replaceSelectionWith(type.create()).scrollIntoView());
      return true;
    });
    return {
      'Shift-Enter': command,
    };
  }
}
```

So if I have a piece of text like this: 

This is just a boring paragraph

When I do a shift+enter with the cursor in the test somewhere I get the following html:

`<p>This is just a boring par<p></p>agraph</p>`

rather than:

`<p>This is just a boring par</p><p>agraph</p>`

Obviously I am just inserting the `<p></p>` node at the caret position. So question is how do I go about getting the results that I want.

Many thanks,
Paul

```

**Comment by 6-connex-eshriner** [comment]
hey paul, how did you disable enter so that it does not create a new paragraph? thanks in advance!

**Comment by andreasvirkus** [comment]
This is the way we've achieved this. Returning true will result in an event being handled, so it won't bubble.

```javascript
new Editor({
  extensions: [
    Extension.create({
      addKeyboardShortcuts() {
        return {
          'Cmd-Enter'() {
            self.$emit('submit')
            return true
          },
          'Ctrl-Enter'() {
            self.$emit('submit')
            return true
          },
        }
      },
    }),
  ]
})
```

Therefore to disable Enter you'd want:

```javascript
Enter() {
  // custom logic here if you'd like
  return true
}
```

**Comment by Voice60** [comment]
Make sure that creating of your new shortcuts placed in array of extensions after starterkit (or extension that have custom shortcuts you are using), otherwise it will not work

Custom shortcuts will work
```
extensions: [
          StarterKit.configure({
            heading: false,
          }),
          Extension.create({
          addKeyboardShortcuts() {
            return {
              'Cmd-Enter'() {
                self.$emit('submit')
                return true
              },
              'Ctrl-Enter'() {
                self.$emit('submit')
                return true
              },
            }
          },
        })]
```

Custom shortcuts will not work
```
extensions: [
          Extension.create({
          addKeyboardShortcuts() {
            return {
              'Cmd-Enter'() {
                self.$emit('submit')
                return true
              },
              'Ctrl-Enter'() {
                self.$emit('submit')
                return true
              },
            }
           },
        }),
          StarterKit.configure({
            heading: false,
          })]
```
